yeonjae/programmers-0/영어가싫어요.js

```diff
@@ -0,0 +1,20 @@
+function solution(numbers) {
+  const strNumberArr = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine",
+  ];
+
+  for (let i = 0; i < strNumberArr.length; i++) {
+    numbers = numbers.split(strNumberArr[i]).join(i);
+  }
+
+  return Number(numbers);
+}
```

Comment:
```suggestion
function solution(numbers) {
  const strNumberArr = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];
  let copyNumbers = numbers;

  for (let i = 0; i < strNumberArr.length; i++) {
    copyNumbers = copyNumbers.split(strNumberArr[i]).join(i);
  }

  return Number(copyNumbers);
}
```
- 풀이 잘 봤습니다 단톡방에서도 언급드렸는데 ㅎㅎ 웬만하면 매개변수 직접 건드리는거 지양하고, 지역변수 생성해서 건드려주세요 ~ (물론 매개변수를 직접 건들이는 상황도 있긴 하지만 안정적인 코딩을 위해서 지역변수로 선언해서 다루는 것을 지향)
- 아래 풀이는 replace와 정규표현식을 활용한 풀이입니다. 참고해주세요
```js
function solution(numbers) {
  const numberMappingObj = {
    zero: 0,
    one: 1,
    two: 2,
    three: 3,
    four: 4,
    five: 5,
    six: 6,
    seven: 7,
    eight: 8,
    nine: 9,
  };
  const regex = /zero|one|two|three|four|five|six|seven|eight|nine/g;
  const replacedNumberMappingStr = numbers.replace(regex, (v) => {
    return numberMappingObj[v];
  });

  return +replacedNumberMappingStr;
}
```

Reply:
넵 아래코드 참고해보겠습니다 !